frontend/components/MapMicro.js

```diff
@@ -1,58 +1,319 @@
-];
+// const TESTREGION = [
+//     [38.8151606, -77.0036513],
+//     [38.8941606, -77.0036513],
+//     [38.8051606, -77.0136513],
+//     [38.8051606, -77.0006513]
+// ];
+
+function sliderInput(value, bound, defaultRange, inputChangeFunc, sliderBlurFunc) {
+    const [minValue, maxValue] = defaultRange;
+
+    return (
+        <Input
+            value={value}
+            margin="dense"
+            onChange={e => inputChangeFunc(e, bound)}
+            onBlur={() => sliderBlurFunc()}
+            inputProps={{
+                "step": 1,
+                "min": minValue,
+                "max": maxValue,
+                "type": "number",
+                "aria-labelledby": "input-slider"
+            }}
+        />
+    );
+}
+
+
+function timeSlider(
+    sliderName, currentRange, defaultRange, lastValid, sliderChangeFunc, inputChangeFunc, sliderBlurFunc
+) {
+    const [minValue, maxValue] = defaultRange;
+    console.log("This is the last valid", lastValid);
+    return (
+        <div key={sliderName}>
+            <Typography id="range-slider" gutterBottom>
+                {sliderName}
+            </Typography>
+            <Grid container spacing={2} alignItems="center">
+                <Grid item>
+                    {sliderInput(
+                        currentRange[0],
+                        "lower",
+                        defaultRange,
+                        inputChangeFunc,
+                        sliderBlurFunc
+                    )}
+                </Grid>
+                <Grid item xs>
+                    <Slider
+                        value={typeof lastValid === "object" ? lastValid : defaultRange}
+                        onChange={(e, v) => sliderChangeFunc(e, v)}
+                        min={minValue}
+                        max={maxValue}
+                        valueLabelDisplay="auto"
+                        aria-labelledby="range-slider"
+                    />
+                </Grid>
+                <Grid item>
+                    {sliderInput(
+                        currentRange[1],
+                        "upper",
+                        defaultRange,
+                        inputChangeFunc,
+                        sliderBlurFunc
+                    )}
+                </Grid>
+            </Grid>
+        </div>
+    );
+}
+
 
 export default class MapMicro extends React.Component {
-    state = {
-        position: [38.9051606, -77.0036513],
-        location: "Deanwood neighborhood, Washington DC"
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            mainLocation: MAIN_LOCATION,
+            markerData: ADDRESS_DATA,
+            sliderState: [1900, 2022],
+            timeRange: [1900, 2022],
+            lastValid: [1900, 2022],
+            names: ["Australia", "Canada", "USA", "Poland", "Spain", "France"]
+        };
     }
 
+    setSliderValue(newLowerBound, newUpperBound) {
+        this.setState({
+            ...this.state,
```

Comment:
I don't think you need to pass in the state object every time to update state properties. Can remove.